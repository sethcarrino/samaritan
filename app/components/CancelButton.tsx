import React from 'react';
import {
  TouchableOpacity,
  Text,
  StyleSheet,
  ViewStyle,
  TextStyle
} from 'react-native';
import { DEVICE_WIDTH } from '../styles/globals';

interface Props {
  text: string;
  onPress: () => void;
}

const CancelButton: React.FC<Props> = ({ text, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

type CancelButtonStyleSheet = {
  container: ViewStyle;
  text: TextStyle;
};

const styles = StyleSheet.create<CancelButtonStyleSheet>({
  container: {
    backgroundColor: '#F25757',
    width: DEVICE_WIDTH - 50,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 16,
    borderRadius: 500,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.4,
    shadowRadius: 3.25,
    elevation: 5
  },
  text: {
    color: '#fff',
    fontSize: 16,
    fontWeight: '600'
  }
});

export default CancelButton;
