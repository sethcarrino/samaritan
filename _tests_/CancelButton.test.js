import React from 'react';
import CancelButton from '../app/components/CancelButton';

import renderer from 'react-test-renderer';

describe('<CancelButton/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CancelButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
