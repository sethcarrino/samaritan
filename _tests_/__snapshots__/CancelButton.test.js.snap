// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CancelButton/> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F25757",
      "borderRadius": 500,
      "elevation": 5,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 16,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.4,
      "shadowRadius": 3.25,
      "width": 700,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  />
</View>
`;
